### PR TITLE
Closes #2:  Automatically pull in data-private updates

### LINKED ISSUE #2: Automate the import of projects from data-private
In order for 18F to make project information available to the public, we should be able to specify project information for projects without repos, and we should be able to add that information to the data that is returned by Team API.

Right now, this is done manually. This story is to automate the import of this data. The logic is currently: anything that exists in both locations will be overwritten by the data in .about.yml.  This is a pairing opportunity.

Acceptance criteria: 

- I am a project owner of a project that doesn't have its own repo
- I add the project to data-private
- The Team API does its regular update of project data
- I should see my project returned when I go to https://team-api.18f.gov/api/projects/

### PR DESCRIPTION
**Pull request by ccostino**
This changeset adds support for pulling in updates from the data-private repo so that when we update the Team API site (team-api.18f.gov) we are retrieving the latest information.  This is important in the cases of when we have projects who do not have their own GitHub repo - the data-private repo is the only source to retrieve information about them.

### CODE REVIEW

**Comment by ccostino** [comment]
I'm putting this up now to help drive conversation over what needs to happen here.  Right now, the changes I've made accomplish the following:

- Adds support in the JSON config file to specify the directory that the data-private repo lives in.
- Adds support to change the working directory that a child process runs in.
- Adds an additional first step to pulling in changes and rebuilding the site; pulling the latest data-private repo changes.

Now that I have this working, I see one major problem.  We want to have the update happen when the data-private repo is updated, which this won't be able to check.  The Team API server is setup to run in the same directory as the repo in which you want to configure web hooks for, and this is currently setup to run for the team-api.18f.gov repo.

In order to get updates from data-private to automatically come through, we need to listen for push events in the data-private repo.  Am I correct in now assuming that this means we'll need the Team API server running in the directory that repo is stored in as well?  Or is there some means of listening to hooks for a different repository with the current setup?

**Comment by ccostino** [comment]
One other thing:  Do we want to focus on just project updates with this, or do we think there is value in listening for any updates in data-private and updating the site accordingly?  If we wire in the npm version of .about.yml validation we now have, I don't see any harm in just updating the site when any change comes through to keep everything current.

File: team-api-config.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
-  "workingDir":   "/home/ubuntu/team-api/",
-  "updateScript": "/home/ubuntu/team-api/scripts/projects/update_project_from_about_yml.rb"
+  "branch": "master",
+  "buildPort": 6000,
+  "updatePort": 6001,
+  "git": "/usr/local/bin/hub",
+  "ruby": "/Users/carlocostino/.rvm/rubies/ruby-2.2.3/bin/ruby",
+  "privateDataDir": "/Users/carlocostino/Projects/data-private",
+  "workingDir": "/Users/carlocostino/Projects/team-api.18f.gov",
+  "updateScript": "/Users/carlocostino/Projects/team-api.18f.gov/scripts/projects/update_project_from_about_yml.rb"
```
Comment (ccostino) [inline team-api-config.json]: As @mbland pointed out, this should not have my local paths in it!

File: index.js [hunk 2]
```diff
@@ -25,7 +25,13 @@ module.exports.launchServer = function(config) {
...
-    ' for .about.yml updates.');
+  hookshot('push', function(info) {
+    var updater = new ProjectDataUpdater(config, info.repository, lock);
+    updater.updateDataPrivate(info);
+  }).listen(config.dataPort);
+
+  console.log('18F Team API - Listening on the following ports:\n' +
+    config.buildPort + ' for push events on ' + config.branch + '\n' +
+    config.updatePort + ' for .about.yml updates\n' +
+    config.updatePort + ' for data-private updates');
```
Comment (ccostino) [inline index.js]: Grrr, copy-paste...

**Comment by ccostino** [comment]
I believe this PR is good to, @mbland and I just walked through it together and made a couple of small adjustments (thank you!).

There is another PR that needs to be made to fully support this, but it will be in the https://github.com/18F/team-api.18f.gov/ as it deals with the update script in there.  We need to modify it to take into account the data-private repo.

**Comment by ccostino** [comment]
A couple more updates are needed:  the submodule should actually go in https://github.com/18F/team-api.18f.gov/ instead of here, and I need to remove an extraneous argument.

**Comment by mbland** [comment]
Looks good to go. Thanks, @ccostino!

**Pull request merged**